tls/s2n_handshake.h

```diff
@@ -141,6 +141,9 @@ struct s2n_handshake {
     /* Indicates the CLIENT_HELLO message has been completely received */
     unsigned client_hello_received:1;
 
+    /* Indicates the HRR message has been received by client */
+    unsigned client_hrr_received:1;
```

Comment:
This may not be necessary. Commented further in `s2n_server_hello_retry_receive`